### PR TITLE
[TACHYON-947] Fix failed security integration test after merging master with PR #1464

### PR DESCRIPTION
**Pull request by dongche**
JIRA: https://tachyon.atlassian.net/browse/TACHYON-947

Fix the failed test `WorkerClientAuthenticationIntegrationTest` and `MasterClientAuthenticationIntegrationTest` by using `AuthenticationFactory` to create the transport in `MasterClientBase`

@apc999 , @ooq , @shenguoquan , @sundapeng , could you please take a look? Thanks

### CODE REVIEW

File: common/src/test/java/tachyon/security/LoginUserTest.java [hunk 1]
```diff
@@ -128,6 +131,9 @@ public void getCustomLoginUserProvidedByAppTest() throws Exception {
 
     Assert.assertNotNull(loginUser);
     Assert.assertEquals(loginUser.getName(), "tachyon-user");
+
+    // clean up
+    System.setProperty(Constants.TACHYON_SECURITY_USERNAME, "");
```
Comment (sundapeng) [inline common/src/test/java/tachyon/security/LoginUserTest.java]: Do you think it is better to use **@After** to the clean work?

**Comment by sundapeng** [comment]
Thank you for your fix, a minor comment left.

File: common/src/test/java/tachyon/security/LoginUserTest.java (hunk 1 above)
Comment (dongche) [inline common/src/test/java/tachyon/security/LoginUserTest.java]: I was also thinking use `@After` at first, but when finding only 2 out of 7 cases need this clean work, just put it in each case.

**Comment by dongche** [comment]
Jenkins, test this please

**Comment by apc999** [comment]
+1 LGTM. Thanks for the quick patch!

**Pull request merged**